mesonbuild/modules/_qt.py

```diff
@@ -477,20 +482,28 @@ def _compile_moc_impl(self, state: ModuleState, kwargs: MocCompilerKwArgs) -> T.
 
         output: T.List[build.GeneratedList] = []
 
+        do_output_json : bool = self._moc_supports_json and kwargs['output_json']
```

Comment:
We always use depfiles and always `mlog.warning()` if they are not supported. For output_json, it's a kwarg so we should error out if it is not supported rather than merely skipping it.

Reply:
I raised an exception here